openedx_authz/views.py

```diff
@@ -0,0 +1,261 @@
+"""
+Views for openedx_authz DRF API.
+"""
+
+from django.shortcuts import get_object_or_404
+from rest_framework import status, viewsets
+from rest_framework.response import Response
+
+from openedx_authz.custom_enforcer import get_enforcer
+
+from .models import Library
+from .serializers import LibrarySerializer
+
+enforcer = get_enforcer()
+
+
+class LibraryViewSet(viewsets.ViewSet):
+    """
+    A ViewSet for handling Library operations using the Library model.
+    Provides all HTTP methods: GET, POST, PUT, PATCH, DELETE
+    """
+
+    def list(self, request):
+        """
+        GET /libraries/
+        Return a list of all libraries.
+        """
+        libraries = Library.objects.all()
+        serializer = LibrarySerializer(libraries, many=True)
+        return Response({"count": libraries.count(), "results": serializer.data})
+
+    def create(self, request):
+        """
+        POST /libraries/
+        Create a new library.
+
+        Example request body:
+
+        ```json
+        {
+            "title": "Title 1",
+            "org": "org1",
+            "slug": "slug1",
+            "description": "Description 1"
+        }
+        ```
+        """
+        serializer = LibrarySerializer(data=request.data)
+        if serializer.is_valid():
+            library = serializer.save()
+            enforcer.add_policy(
+                self.request.user.username,
+                f"{self.request.path}{library.id}/",
+                "(GET)|(PUT)|(DELETE)|(PATCH)",
+            )
```

Comment:
I think these kind of operations should be decoupled from the application code, like event-based. What do you think?